SR2 Community Patch/scripts/server/objects/Civilian.as

```diff
@@ -434,6 +456,46 @@ tidy class CivilianScript {
 		return 0.2;
 	}
 
+	bool arriveWithEmpirePropulsionTechnology(Civilian& obj, vec3d enterDest) {
+		if(awaitingGateJump)
+			return obj.moveTo(enterDest, moveId, enterOrbit=false);
+
+		if(obj.owner.hasTrait(getTraitID("Jumpdrive"))) {
+			playParticleSystem("GateFlash", obj.position, obj.rotation, obj.radius, obj.visibleMask);
+			awaitingGateJump = true;
+			obj.position = enterDest;
+			obj.clearMovement();
+			playParticleSystem("GateFlash", enterDest, obj.rotation, obj.radius, obj.visibleMask);
+			return true;
+		}
+		if(obj.owner.hasTrait(getTraitID("Hyperdrive"))) {
+			if(!obj.inFTL)
+				playParticleSystem("GateFlash", obj.position, obj.rotation, obj.radius, obj.visibleMask);
+			if(enterDest.distanceToSQ(obj.position) < DEST_RANGE * DEST_RANGE ||
+				obj.FTLTo(enterDest, ACC_INTERSYSTEM * 4, moveId)
+			) {
+				obj.FTLDrop();
+				playParticleSystem("GateFlash", obj.position, obj.rotation, obj.radius, obj.visibleMask);
+				return true;
+			}
+			return false;
+		}
+		return obj.moveTo(enterDest, moveId, enterOrbit=false);
+	}
+
+	bool hasGateToNextRegion(Region& curRegion, Empire& owner) {
+		if(hasOddityLink(curRegion, nextRegion))
```

Comment:
Does this result in the civillian taking the closest oddity/gate to them and their target if there are multiple?

Reply:
Also what happens if the slipstream is about to close before the civillian can reach it?

Reply:
That should, AFAIK, result in the tear being ignored by the civilian ship, just like a military one.

Reply:
Nope, this is just for the civilians to not go to the system border, when they want to leave. Without this, they `move to the system border to warp off` and then `warp off in the direction of the portal, in that system`.

This doesn't change any pathing, trade will still not work through ss unless you have a Commerce Station in the other trade area. Ships _will_ only use wh/ss since their pathfinder will take the shortest path without them knowing about it.

The normal slipstream pathing doesn't consider timers below 60s, this ignores the timer but it also only considers portals in the local system, not anywhere else, like the usual pathing. 

If a ship happens to take a path through a closing portal and it closes before it reaches the portal, the ship re-paths the moment the portal closes, which can lead to it warping off mid-system. But that should be rare.